src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/RcaController.java

```diff
@@ -255,6 +267,47 @@ private void readRcaEnabledFromConf() {
         });
   }
 
+  /**
+   * Reads the mutedRCAList value from the rca.conf file, performs validation on the param value
+   * provided and on successful validation, updates the AnalysisGraph with muted RCA value.
+   *
+   * <p>In case all the RCAs in param value are incorrect, return without any update.
+   */
+  private void readAndUpdateMutesRcas() {
+    // If the rca config file has been updated since the lastModifiedTimeInMillisInMemory in memory,
+    // refresh the `muted-rcas` value from rca config file.
+    final RcaConf rcaConf = RcaControllerHelper.pickRcaConfForRole(currentRole);
+    long lastModifiedTimeInMillisOnDisk = rcaConf.getLastModifiedTime();
+    if (lastModifiedTimeInMillisOnDisk > lastModifiedTimeInMillisInMemory) {
+      try {
+        List<String> rcasForMute = rcaConf.getMutedRcaList();
+        LOG.info("RCAs provided for muting : {}", rcasForMute);
+
+        Set<String> graphNodeNames = new HashSet<>();
+        RcaUtil.getAnalysisGraphComponents(rcaConf).forEach(
+                connectedComponent -> graphNodeNames.addAll(connectedComponent.getNodeNames()));
+
+        Set<String> validRcasForMute = rcasForMute.stream()
+                .filter(rcaForMute -> graphNodeNames.contains(rcaForMute))
+                .collect(Collectors.toSet());
+
+        // If validRcasForMute is empty but rcasForMute is not empty
+        // all the input RCAs are incorrect, return.
+        if (validRcasForMute.isEmpty() && !rcasForMute.isEmpty()) {
+          LOG.info("Incorrect RCA value, cannot be muted : {}", rcasForMute);
+          return;
+        }
+
+        LOG.info("Updating the muted RCA Graph to : {}", validRcasForMute);
+        Stats.getInstance().updateMutedGraphNodes(validRcasForMute);
+      } catch (Exception e) {
+        LOG.error("Couldn't read/update the muted RCAs. Ran into {}", e.getMessage());
+        e.printStackTrace();
```

Comment:
Instead of this, we should add this to the log ?
LOG.error("Couldn't read/update the muted RCAs.", e);

Reply:
Yup, updated this.